data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/lightclient/GetLightClientUpdatesByRange.java

```diff
@@ -73,4 +86,41 @@ public void handleRequest(RestApiRequest request) throws JsonProcessingException
 
     return SerializableTypeDefinition.listOf(lightClientUpdateObjectType);
   }
+
+  private static ResponseContentTypeDefinition<List<ObjectAndMetaData<LightClientUpdate>>>
+      getSszResponseType(SchemaDefinitionCache schemaDefinitionCache) {
+    OctetStreamResponseContentTypeDefinition.OctetStreamSerializer<
+            List<ObjectAndMetaData<LightClientUpdate>>>
+        serializer =
+            (data, out) -> {
+              LightClientUpdateResponseSchema schema =
+                  SchemaDefinitionsAltair.required(
+                          schemaDefinitionCache.getSchemaDefinition(SpecMilestone.ALTAIR))
+                      .getLightClientUpdateResponseSchema();
+
+              data.stream()
+                  .forEach(
+                      lightClientUpdateObjectAndMetaData -> {
+                        // Get the fork version for the slot of `update.attested_header`
+                        Bytes4 forkDigest = Bytes4.fromHexString("TODO");
```

Comment:
For this here we're looking at just the definition of the object for metadata purposes, would expect something similar to `v2.debug.GetState.getResponseType`, and then sszResponseType should be able to take the same 'data' object and serialize it as ssz...
It may be more complicated if the `data` is not consistent with what's sent in octet-stream, but it's a good concrete example...

Reply:
Thanks @rolfyone, I gave it another pass and simplified the interface.

Indeed this endpoint is tricky, the JSON serialization and octet serialization are effectively two different types: https://github.com/ethereum/beacon-APIs/blob/master/apis/beacon/light_client/updates.yaml#L26

For this reason I propose returning a JSON type of `List<ObjectAndMetadata<LightClientUpdate>>` and an SSZ type of `List<LightClientUpdateResponse>` (where LCUResponse is the wrapper object with fork digest, etc.) I've implemented it this way in the latest commit.

Later when we implement the endpoint, we will simply compute the fork digest in the business logic (i.e. `ChainDataProvider`) rather than trying to compute it in the endpoint... this is more sensible anyway because fork digest needs the fork version, which is located in `BeaconState`. Concretely, I will expect `ChainDataProvider` to have a function for light client updates with return type `List<ObjectAndMetadata<LightClientUpdateResponse>>`. For the JSON type, we can pull out the `version` and the `LightClientUpdate`. For the SSZ type, we simply take the `LightClientUpdateResponse`. Maybe in the end it will look slightly different, but at least this way should work with the interface we specify in this PR.

Reply:
Sounds like a good plan to start with, at the very least :)